repository-data/webfiles/src/main/resources/site/freemarker/common/macro/sections/tableauSection.ftl

```diff
@@ -0,0 +1,34 @@
+<#ftl output_format="HTML">
+<#include "../../../include/imports.ftl">
+<#-- @ftlvariable name="section" type="uk.nhs.digital.website.beans.Tableau" -->
+
+<#macro tableau section index>
+
+    <#if section?? && section.url??>
+        <@hst.headContribution>
+            <script type="text/javascript"
+                    src="https://public.tableau.com/javascripts/api/tableau-2.min.js"></script>
+        </@hst.headContribution>
+
+        <#assign divId = "tableau-${index}"/>
+
+        <div id="${divId}"></div>
+
+        <script type="text/javascript">
+            window.onload(initViz());
```

Comment:
The `window.loaded` loaded event is limited to just one per page. This therefore could conflict with other scripts. You should use `window.addEventListener('load', initViz);` instead as this event is not limited to one.

Reply:
Added eventListener to invoke the function trigger